package/MDAnalysis/analysis/msd.py

```diff
@@ -421,3 +427,40 @@ def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
                 positions[:, n, :]
             )
         self.results.timeseries = self.results.msds_by_particle.mean(axis=1)
+    
+
+    def _conclude_non_linear(self):
+        
+        dump_times = [ts.time for ts in self._trajectory]
+        n_frames = len(dump_times)
```

Comment:
Just use `self.nframes`
```suggestion
        n_frames = self.n_frames
```

Sorry, that's not properly documented but it is created with [AnalysisBase._prepare_sliced_trajectory](https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html#MDAnalysis.analysis.base.AnalysisBase._prepare_sliced_trajectory) (I just updated the docs ... #5075 )